lib/app/widgets/top_level_screen_wrapper.dart

```diff
@@ -0,0 +1,93 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:logger_flutter/logger_flutter.dart';
+import 'package:schulcloud/sign_in/sign_in.dart';
+
+import '../logger.dart';
+import '../services/storage.dart';
+import '../utils.dart';
+import 'schulcloud_app.dart';
+
+class TopLevelScreenWrapper extends StatefulWidget {
+  const TopLevelScreenWrapper({@required this.child}) : assert(child != null);
+
+  final Widget child;
+
+  @override
+  _TopLevelScreenWrapperState createState() => _TopLevelScreenWrapperState();
+}
+
+class _TopLevelScreenWrapperState extends State<TopLevelScreenWrapper> {
+  StreamSubscription _deepLinksSubscription;
+  Uri _uriToVisit;
+
+  @override
+  void initState() {
+    super.initState();
+
+    // Not the prettiest solution, but it works 😇
+    _deepLinksSubscription = _subscribe().listen((_) => setState(() {}));
+  }
+
+  @override
+  void dispose() {
+    _deepLinksSubscription.cancel();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return LogConsoleOnShake(
+      child: Scaffold(
+        body: Builder(builder: (context) {
+          scheduleMicrotask(() => _handleUri(context));
```

Comment:
While I could replace `BuildContext` with a concrete scaffold and `S`, we still need a reference to it for the `signOut`-function that shows a dialog (which is not possible without a `BuildContext`)… Any idea on how to improve this?

Reply:
What about yielding the URI to visit in the `_subscribe` method and then, in `initState`, doing
```
_deepLinksSubscription = _subscribe.listen(_handleUri);
```
The `_handleUri` could just take a `Uri` because it already has a `context` because it's in the `State` of the widget.
Also, I remember we taked about this but what was our reasoning again to use `StreamQueue` instead of a simple `BroadcastStream`?

Reply:
A `BroadcastStream` would swallow the initial URI and a `BehaviorSubject` would handle some URIs twice when switching between SignIn and SignedIn.

Direct handling of the URI should work, but I'll have to wait for our `SnackBar`-EventQueue in #207 as a `SnackBar` gets shown when trying to open a link while not signed in.